drivers/pcie/host/pcie.c

```diff
@@ -49,55 +49,68 @@ void pcie_set_cmd(pcie_bdf_t bdf, uint32_t bits, bool on)
 
 bool pcie_get_mbar(pcie_bdf_t bdf, unsigned int index, struct pcie_mbar *mbar)
 {
-	bool valid_bar = false;
+	uintptr_t phys_addr;
 	uint32_t reg;
-	uintptr_t phys_addr = PCIE_CONF_BAR_NONE;
-	size_t size = 0;
+	size_t size;
 
 	for (reg = PCIE_CONF_BAR0; reg <= PCIE_CONF_BAR5; reg++) {
-		uint32_t addr = pcie_conf_read(bdf, reg);
+		uintptr_t addr = pcie_conf_read(bdf, reg);
```

Comment:
This could be directly read into phys_addr. This way we do not have to read the address of the searched mbar twice.

Reply:
true